src/utils/URLUtil.js

```diff
@@ -0,0 +1,51 @@
+import util from './Util';
+import parseUri from 'parseUri';
+import containerActions from '../actions/ContainerActions';
+
+module.exports = {
+  TYPE_WHITELIST: ['repository'],
+  METHOD_WHITELIST: ['run'],
+  openUrl: function (url, flags, appVersion) {
+    if (!url || !flags || !flags.dockerURLEnabledVersion || !appVersion) {
+      return false;
+    }
+
+    // Make sure this feature is enabled via the feature flag
+    if (util.compareVersions(appVersion, flags.dockerURLEnabledVersion) < 0) {
+      return false;
+    }
+
+    var parser = parseUri(url);
+
+    if (parser.protocol !== 'kitematic') {
+      return false;
+    }
+
+    // Get the type of object we're operating on, e.g. 'repository'
+    var type = parser.host;
+
+    if (this.TYPE_WHITELIST.indexOf(type) === -1) {
+      return false;
+    }
+
+    // Separate the path into [run', 'redis']
+    var tokens = parser.path.replace('/', '').split('/');
```

Comment:
Since Kitematic only supports Docker Hub repos, I think it's best that we limit to Docker Hub repos (for now), using a check like this:

```
if (tokens.length > 3) {
  return false;
}
```
to avoid pulling from other repos. For example

`['run', 'kitematic', 'hello-world-nginx'] // kitematic/hello-world-nginx` is ok but
`['run', 'other-registry-host.com', 'namespace', 'reponame'] // other-registry-host.com/namespace/reponame` should not be accepted for now